src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -186,19 +186,21 @@ public async Task EmissionsForecastsBatch_MissingRequiredParams_ReturnsBadReques
         Assert.That(result?.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
     }
 
-    [TestCase("2021-09-01T08:30:00Z", "2021-09-01T09:00:00Z", "2021-09-01T12:00:00Z", "eastus", 1, TestName = "EmissionsForecastsBatch expects OK for eastus 1 element")]
-    [TestCase("2021-09-01T08:30:00Z", "2021-09-01T08:30:00Z", "2021-09-02T08:30:00Z", "westus", 3, TestName = "EmissionsForecastsBatch expects OK for westus 3 element")]
-    public async Task EmissionsForecastsBatch_SupportedDataSources_ReturnsOk(DateTimeOffset reqAt, DateTimeOffset start, DateTimeOffset end, string location, int nelems)
+    [TestCase("2021-09-01T08:30:00Z", "2021-09-01T09:00:00Z", "2021-09-01T12:00:00Z", "eastus", 1, TestName = "EmissionsForecastsBatch expects OK for single element")]
+    [TestCase("2021-09-01T08:30:00Z", "2021-09-01T08:30:00Z", "2021-09-02T08:30:00Z", "westus", 3, TestName = "EmissionsForecastsBatch expects OK for multiple elements")]
+    public async Task EmissionsForecastsBatch_SupportedDataSources_ReturnsOk(string reqAt, string start, string end, string location, int nelems)
     {
         IgnoreTestForDataSource("data source does not implement '/emissions/forecasts/batch'", DataSourceType.JSON);
 
+        var expectedRequestedAt = DateTimeOffset.Parse(reqAt);
+        var expectedDataStartAt = DateTimeOffset.Parse(start);
+        var expectedDataEndAt = DateTimeOffset.Parse(end);
         _dataSourceMocker.SetupBatchForecastMock();
-        var inputData = Enumerable.Range(0, nelems).Select(x => new EmissionsForecastBatchDTO() 
-        {
-            RequestedAt = reqAt,
-            DataStartAt = start,
-            DataEndAt = end,
-            Location = location
+        var inputData = Enumerable.Range(0, nelems).Select(x => new {
+            requestedAt = expectedRequestedAt,
```

Comment:
The test payload should use our string inputs directly
```suggestion
            requestedAt = reqAt,
```